multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -138,4 +154,55 @@ private FileMetadata parseFileEntry(FileEntry fileEntry) {
                                     .namespace(fileEntry.getNamespace())
                                     .build();
     }
+
+    private FileEntry uploadFileFromUrl(String spaceGuid, String namespace, String mtarEndpoint) throws FileStorageException, IOException {
+        if (!UriUtil.isValidUrl(mtarEndpoint)) {
+            throw new SLException("Invalid URL");
```

Comment:
Add the URL to the exception and make sure that the response contains the error message, so that users will be able to see what went wrong.